dfdaemon/downloader/dfget/dfget.go

```diff
@@ -80,50 +59,45 @@ func (dfGetter *DFGetter) Download(url string, header map[string][]string, name
 	return "", fmt.Errorf("dfget fail(%s):%v", cmd.ProcessState.String(), err)
 }
 
-func (dfGetter *DFGetter) parseCommand(url string, header map[string][]string, name string) (
-	cmdPath string, args []string, dstPath string) {
-	args = make([]string, 0, 32)
-	args = append(append(args, "-u"), url)
-	args = append(append(args, "-o"), dfGetter.dstDir+name)
-	if dfGetter.notbs {
-		args = append(args, "--notbs")
+// getCommand returns the command to download the given resource
+func (dfGetter *DFGetter) getCommand(
+	url string, header map[string][]string, output string,
+) (cmd *exec.Cmd) {
+	args := []string{
+		"--dfdaemon",
+		"-u", url,
+		"-o", output,
 	}
 
-	if strings.TrimSpace(dfGetter.callSystem) != "" {
-		args = append(append(args, "--callsystem"), strings.TrimSpace(dfGetter.callSystem))
-	}
-	if strings.TrimSpace(dfGetter.urlFilter) != "" {
-		args = append(append(args, "-f"), strings.TrimSpace(dfGetter.urlFilter))
-	}
-	if strings.TrimSpace(dfGetter.rateLimit) != "" {
-		args = append(append(args, "-s"), dfGetter.rateLimit)
-		args = append(append(args, "--totallimit"), dfGetter.rateLimit)
-	}
-	if dfGetter.supernodes != nil && len(dfGetter.supernodes) > 0 {
-		args = append(append(args, "--node"), strings.Join(dfGetter.supernodes, ","))
+	if dfGetter.config.Notbs {
+		args = append(args, "--notbs")
 	}
 
-	if header != nil {
-		for key, value := range header {
-			// discard HTTP host header for backing to source successfully
-			if strings.EqualFold(key, "host") {
-				continue
-			}
-			if value != nil && len(value) > 0 {
-				for _, oneV := range value {
-					args = append(append(args, "--header"), fmt.Sprintf("%s:%s", key, oneV))
-				}
-			} else {
-				args = append(append(args, "--header"), fmt.Sprintf("%s:%s", key, ""))
-			}
-
+	add := func(key, value string) {
+		if v := strings.TrimSpace(value); v != "" {
```

Comment:
Reusing functions is good, and the changed code would look like this: 

```go
        if !common.IsEmptyStr(value) {
                args = append(args, key, strings.TrimSpace(value))
        }
```

I would argue that the current implementation looks a little cleaner. And since we use the result of `strings.TrimSpace` afterwards, not only for checking if `value` is empty, it's actually fine.

Also I think we might not need to trim spaces here in the first place, because the values all come from cli parameters or config file, where extra spaces would be handled during parsing. What is the original purpose of trimming spaces here? Maybe this would be enough

```go
        if value != "" {
                args = append(args, key, value)
        }
```